src/main/java/duke/utils/Parser.java

```diff
@@ -0,0 +1,149 @@
+package duke.utils;
+
+import duke.exceptions.EmptyBodyException;
+import duke.exceptions.InvalidKeyException;
+import duke.exceptions.InvalidNumberException;
+import duke.exceptions.WrongFormatException;
+
+/**
+ * Parser to read and understand user inputs. Stores the parsed information inside the object.
+ */
+public class Parser {
+	private KeyEnum currentKey = KeyEnum.INVALID;
+	private String inputDetail;
+	private String to;
+	private String from;
+	private Integer index;
+
+    /**
+     * Parses user input and store the information.
+     * @param userInput String user input.
+     * @throws InvalidKeyException If the command keyword is unknown.
+     * @throws EmptyBodyException If the command body is empty.
+     * @throws WrongFormatException If the format of command is wrong.
+     * @throws InvalidNumberException If the number in the command is not a number.
+     */
+    public void parse(String userInput) throws InvalidKeyException, EmptyBodyException, WrongFormatException, InvalidNumberException {
+        String[] userInputSplit = userInput.split(" ");
+        this.determineCurrentKey(userInputSplit[0]);
+        switch (this.currentKey) {
+            case DEADLINE:
```

Comment:
Should the case clauses have indentations? I noticed the same issue in several other places too.